integration/kotlinx-coroutines-jdk8/src/future/Future.kt

```diff
@@ -70,7 +71,11 @@ private class CompletableFutureCoroutine<T>(
  */
 public fun <T> Deferred<T>.asCompletableFuture(): CompletableFuture<T> {
     val future = CompletableFuture<T>()
-    future.whenComplete { _, exception -> cancel(exception) }
+    future.whenComplete { _, exception ->
+        cancel(exception?.let {
+            it as? CancellationException ?: CancellationException("Future failed", it)
```

Comment:
It happens when CompletableFuture has been completed exceptionally (and we cancelling deferred because of that). I've clarified the message correspondingly.